pymeasure/instruments/agilent/agilent33500.py

```diff
@@ -48,138 +46,462 @@
+                'The binary format has not yet been implemented. Use "DAC" or "float" instead.'
+            )
+        else:
+            raise ValueError(
+                'Undefined format keyword was used. Valid entries are "DAC", "float" and "binary"'
+            )
+
+
 class Agilent33500(Instrument):
-    """Represents the Agilent 33500 Function/Arbitrary Waveform Generator family.
+    """
+    Represents the Agilent 33500 Function/Arbitrary Waveform Generator family.
+
     Individual devices are represented by subclasses.
+    User can specify a channel to control, if no channel specified, a default channel
+    is picked based on the device e.g. For Agilent33500B the default channel
+    is channel 1. See reference manual for your device
 
     .. code-block:: python
 
         generator = Agilent33500("GPIB::1")
 
-        generator.shape = 'SIN'                 # Sets the output signal shape to sine
-        generator.frequency = 1e3               # Sets the output frequency to 1 kHz
-        generator.amplitude = 5                 # Sets the output amplitude to 5 Vpp
-        generator.output = 'on'                 # Enables the output
+        generator.shape = 'SIN'                 # Sets default channel output signal shape to sine
+        generator.ch[1].shape = 'SIN'           # Sets channel 1 output signal shape to sine
+        generator.frequency = 1e3               # Sets default channel output frequency to 1 kHz
+        generator.ch[1].frequency = 1e3         # Sets channel 1 output frequency to 1 kHz
+        generator.ch[2].amplitude = 5           # Sets channel 2 output amplitude to 5 Vpp
+        generator.ch[2].output = 'on'           # Enables channel 2 output
 
-        generator.shape = 'ARB'                 # Set shape to arbitrary
-        generator.arb_srate = 1e6               # Set sample rate to 1MSa/s
+        generator.ch[1].shape = 'ARB'           # Set channel 1 shape to arbitrary
+        generator.ch[1].arb_srate = 1e6         # Set channel 1 sample rate to 1MSa/s
 
-        generator.data_volatile_clear()         # Clear volatile internal memory
-        generator.data_arb(                     # Send data points of arbitrary waveform
+        generator.ch[1].data_volatile_clear()   # Clear channel 1 volatile internal memory
+        generator.ch[1].data_arb(               # Send data of arbitrary waveform to channel 1
             'test',
             range(-10000, 10000, +20),          # In this case a simple ramp
             data_format='DAC'                   # Data format is set to 'DAC'
-        )
-        generator.arb_file = 'test'             # Select the transmitted waveform 'test'
+         )
+        generator.ch[1].arb_file = 'test'       # Select the transmitted waveform 'test'
 
     """
 
-    id = Instrument.measurement(
-        "*IDN?", """ Reads the instrument identification """
-    )
+    ch = Instrument.ChannelCreator(Agilent33500Channel, (1, 2))
```

Comment:
Just for your information:
The ChannelCreator will add also the channels as `inst.ch_1` and `inst.ch_2` (being an alternative to `inst.ch[1]`).

Reply:
Thank you for the comment